lib/migration.pm

```diff
@@ -110,5 +111,34 @@ sub record_disk_info {
     assert_script_run 'df -h > /tmp/df.txt';
     upload_logs '/tmp/df.txt';
 }
+
+# System check after snapper rollback
+sub check_rollback_system {
+    # Check if repos are rolled back to correct version
+    script_run("zypper lr -u | tee /dev/$serialdev");
+    my $incorrect_repos = script_output("
+        version=\$(grep VERSION= /etc/os-release | cut -d'=' -f2 | cut -d' ' -f1 | sed 's/\"//g')
+        base_version=\$(echo \$version | cut -d'-' -f1)
+        zypper lr | cut -d'|' -f3 | gawk '/SLE/ || /openSUSE/' | sed \"/\$version\\|Module.*\$base_version/d\"
+    ", 100);
+    record_info('Incorrect Repos', $incorrect_repos, result => 'fail') if $incorrect_repos;
+
+    # Check SUSEConnect status for SLE
+    if (is_sle) {
+        # check rollback-helper service is enabled and worked properly
+        my $output = script_output "systemctl status rollback.service";
+        if ($output !~ /enabled.*?code=exited,\sstatus=0\/SUCCESS/s) {
+            die "rollback service was failed";
+        }
+        # system is un-registered during media based upgrade
+        unless (get_var('MEDIA_UPGRADE')) {
+            # Verify registration status matches current system version
+            assert_script_run 'curl -O ' . data_url('console/check_registration_status.py');
+            assert_script_run 'chmod +x check_registration_status.py';
+            assert_script_run './check_registration_status.py';
```

Comment:
I guess you can simplify these three lines to `assert_script_run '$(curl -s ' . data_url('console/check_registration_status.py') . ')';`

Reply:
Changed to:
assert_script_run('$(curl -s ' . data_url('console/check_registration_status.py') . ')') unless get_var('MEDIA_UPGRADE');

and testing the updates now.

Reply:
This doesn't work as expected, i.e. http://openqa-apac1.suse.de/tests/685#step/snapper_rollback/39